src/latex/song2tex.py

```diff
@@ -109,48 +115,83 @@ def parseDOM(root, linked=False):
         text_blocks = root.findall('{*}lyric/{*}block')
-        flatten = lambda block: Block.parseDOM(block) if 'blocknb' not in block.attrib else Block.parseDOM(text_blocks[int(block.attrib['blocknb'])-1], linked=True)
-        blocks = [flatten(block) for block in root.find('{*}lyric').getchildren() if block.tag != '{http://21wdh.staszic.waw.pl}tabbs']
+        flatten = lambda block: Block.parseDOM(block) if 'blocknb' not in block.attrib else Block.parseDOM(
+            text_blocks[int(block.attrib['blocknb']) - 1], linked=True)
+        blocks = [flatten(block) for block in root.find('{*}lyric').getchildren() if
+                  block.tag != '{http://21wdh.staszic.waw.pl}tabbs']
         get_text = lambda elem: elem.text if elem is not None else None
         if blocks and blocks[-1].rows:
           blocks[-1].rows[-1].row_type += 'E' # end row
 
         return Song(
-            title = root.get('title'),
-            text_author = get_text(root.find('{*}text_author')),
-            composer = get_text(root.find('{*}composer')),
-            artist = get_text(root.find('{*}artist')),
-            blocks = blocks
+            title=root.get('title'),
+            text_author=get_text(root.find('{*}text_author')),
+            composer=get_text(root.find('{*}composer')),
+            artist=get_text(root.find('{*}artist')),
+            blocks=blocks
         )
 
-tree = etree.parse(sys.argv[1])
-song = Song.parseDOM(tree.getroot())
-
-latex_jinja_env = jinja2.Environment(
-    block_start_string = '\BLOCK{',
-    block_end_string = '}',
-    variable_start_string = '\VAR{',
-    variable_end_string = '}',
-    comment_start_string = '\#{',
-    comment_end_string = '}',
-    line_statement_prefix = '%%',
-    line_comment_prefix = '%#',
-    trim_blocks = True,
-    lstrip_blocks = True,
-    autoescape = False,
-    loader = jinja2.FileSystemLoader(sys.path[0])
-)
-template = latex_jinja_env.get_template('song_template.tex')
-print(template.render(song=song))
+
+def s2t(path):
+    path = path
+    tree = etree.parse(path)
+    song = Song.parseDOM(tree.getroot())
+
+    latex_jinja_env = jinja2.Environment(
+        block_start_string='\BLOCK{',
+        block_end_string='}',
+        variable_start_string='\VAR{',
+        variable_end_string='}',
+        comment_start_string='\#{',
+        comment_end_string='}',
+        line_statement_prefix='%%',
+        line_comment_prefix='%#',
+        trim_blocks=True,
+        lstrip_blocks=True,
+        autoescape=False,
+        loader=jinja2.FileSystemLoader(sys.path[0])
+    )
+    template = latex_jinja_env.get_template('song_template.tex')
+    return template.render(song=song)
+
+
+def main():
+    if len(sys.argv) < 2:
+        print("Wymagana nazwa pliku xml", file=sys.stderr)
+        exit(1)
+    tree = etree.parse(sys.argv[1])
+    song = Song.parseDOM(tree.getroot())
+
+    latex_jinja_env = jinja2.Environment(
```

Comment:
Let's reuse song2tex method instead of duplicating the logic.

Reply:
Done